src/game/Objects/Unit.cpp

```diff
@@ -276,7 +276,7 @@ void Unit::Update(uint32 update_diff, uint32 p_time)
 
     // extra attack
     Unit* victim = GetVictim();
-    if (IsInCombat() && m_doExtraAttacks && GetExtraAttacks() && victim && (CanAutoAttackTarget(victim) == ATTACK_RESULT_OK))
+    if (IsInCombat() && m_doExtraAttacks && GetExtraAttacks() && victim && (CanAutoAttackTarget(victim) == ATTACK_RESULT_OK) && !IsNonMeleeSpellCasted(false, false, true))

```

Comment:
This allows extra attacks to be applied after spell casting and channeling are complete, which is the behaviour in classic.